### PR TITLE
feat(server): Implement from domain models to Adapter on the public

### PR DESCRIPTION
**Pull request by gop50k**
# Overview
Implementation of Public APIs
From domain models to Adapter, implemented in project and model.

# Ticket
https://github.com/reearth/reearth-cms/issues/28

### CODE REVIEW

File: server/internal/adapter/gql/resolver_mutation_model.go [hunk 1]
```diff
@@ -13,12 +13,13 @@ func (r *mutationResolver) CreateModel(ctx context.Context, input gqlmodel.Creat
...
 		return nil, err
 	}
-
+	var m *interfaces.CreateModelParam
 	res, err := usecases(ctx).Model.Create(ctx, interfaces.CreateModelParam{
 		ProjectId:   pId,
 		Name:        input.Name,
 		Description: input.Description,
 		Key:         input.Key,
+		Public:      m.Public,
```
Comment (rot1024) [inline server/internal/adapter/gql/resolver_mutation_model.go:22]:
```suggestion
		Public:      false,
```

File: server/internal/adapter/gql/resolver_mutation_model.go [hunk 2]
```diff
@@ -35,11 +36,13 @@ func (r *mutationResolver) UpdateModel(ctx context.Context, input gqlmodel.Updat
 		return nil, err
 	}
 
+	var m *interfaces.UpdateModelParam
 	res, err := usecases(ctx).Model.Update(ctx, interfaces.UpdateModelParam{
 		ModelId:     mId,
 		Name:        input.Name,
 		Description: input.Description,
 		Key:         input.Key,
+		Public:      m.Public,
```
Comment (rot1024) [inline server/internal/adapter/gql/resolver_mutation_model.go:45]:
```suggestion
		Public:      input.Public,
```

File: server/internal/usecase/interactor/model.go [hunk 3]
```diff
@@ -80,6 +80,9 @@ func (i Model) Create(ctx context.Context, param interfaces.CreateModelParam, op
 			if param.Description != nil {
 				mb = mb.Description(*param.Description)
 			}
+			if param.Public != nil {
+				mb = mb.IsPublic(*param.Public)
```
Comment (rot1024) [inline server/internal/usecase/interactor/model.go:84]: Rename IsPablic to Public

File: server/pkg/model/model_test.go [hunk 4]
```diff
@@ -214,12 +214,19 @@ func TestModel_Public(t *testing.T) {
 		want  bool
 	}{
 		{
-			name: "test",
+			name: "set public true",
```
Comment (rot1024) [inline server/pkg/model/model_test.go:217]:
```suggestion
			name: "public true",
```

Confusing name; nothing is set.

File: server/pkg/model/model_test.go [hunk 5]
```diff
@@ -214,12 +214,19 @@ func TestModel_Public(t *testing.T) {
...
 			model: Model{
 				public: true,
 			},
 			want: true,
 		},
+		{
+			name: "set public false",
```
Comment (rot1024) [inline server/pkg/model/model_test.go:224]:
```suggestion
			name: "public false",
```

File: server/pkg/model/model_test.go [hunk 6]
```diff
@@ -374,7 +381,7 @@ func TestModel_SetPublic(t *testing.T) {
 		args args
 	}{
 		{
-			name: "test",
+			name: "set public true",
 			args: args{
 				public: true,
```
Comment (rot1024) [inline server/pkg/model/model_test.go:386]: public: false

We want to test if the field is correctly changed:

True -> false or false -> true

File: server/pkg/model/model_test.go [hunk 7]
```diff
@@ -383,7 +390,7 @@ func TestModel_SetPublic(t *testing.T) {
 			},
 		},
 		{
-			name: "test",
+			name: "set public faalse",
 			args: args{
 				public: false,
```
Comment (rot1024) [inline server/pkg/model/model_test.go:395]: public: true

File: server/internal/adapter/gql/resolver_mutation_model.go [hunk 8]
```diff
@@ -13,12 +13,13 @@ func (r *mutationResolver) CreateModel(ctx context.Context, input gqlmodel.Creat
 	if err != nil {
 		return nil, err
 	}
-
+	var m *interfaces.CreateModelParam
```
Comment (rot1024) [inline server/internal/adapter/gql/resolver_mutation_model.go:16]: Can you please explain why this is necessary? Seems like an unnecessary change to me.

**Review by rot1024 — CHANGES_REQUESTED** [review]

File: server/internal/adapter/gql/resolver_mutation_model.go (hunk 8 above)
Comment (gop50k) [reply server/internal/adapter/gql/resolver_mutation_model.go:16]: @rot1024 
I thought it was not a good idea to add a Public field to an already generated CreateModelParam, so I set it as a variable in advance.

```
input is defined like this
input gqlmodel.CreateModelInput
type CreateModelInput struct {
	ProjectID   ID      `json:"projectId"`
	Name        *string `json:"name"`
	Description *string `json:"description"`
	Key         *string `json:"key"`
}


input gqlmodel.UpdateModelInput
type UpdateModelInput struct {
	ModelID     ID      `json:"modelId"`
	Name        *string `json:"name"`
	Description *string `json:"description"`
	Key         *string `json:"key"`
}
```

File: server/internal/adapter/gql/resolver_mutation_model.go [hunk 9]
```diff
@@ -6,19 +6,20 @@ import (
...
 	if err != nil {
 		return nil, err
 	}
-
 	res, err := usecases(ctx).Model.Create(ctx, interfaces.CreateModelParam{
 		ProjectId:   pId,
 		Name:        input.Name,
 		Description: input.Description,
 		Key:         input.Key,
+		Public:      lo.ToPtr(false),
```
Comment (rot1024) [inline server/internal/adapter/gql/resolver_mutation_model.go:22]:
```suggestion
		Public:      nil,
```

nil is ok

**Review by rot1024 — APPROVED** [review]
